tests/test_training/test_trainer.py

```diff
@@ -20,11 +20,16 @@
 import pytest
 
 import numpy as np
+import ray
 from mrmustard.lab import Vacuum, Dgate, Ggate, Gaussian
 from mrmustard.physics import fidelity
 from mrmustard.training import Optimizer
 from mrmustard.training.trainer import map_trainer, train_device, update_pop
 
+NUM_CPUS = 1
+ray.init(num_cpus=NUM_CPUS)
+print(f"RAY INITIATED: {ray.is_initialized()}")
```

Comment:
I think pytest supresses print statements, I believe using a logger is better — in that way pytest can capture and handle the logs accordingly.
```suggestion
logging.info(f"RAY INITIATED: {ray.is_initialized()}")
```
Don't forget to import the logging module.

Reply:
Nice catch, this was used for debugging with `pytest -s`, no point for it being here anymore, will just delete.